components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -706,47 +732,231 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
+                    if (isUserConsentExist) {
+                        oAuth2ScopeService.updateUserConsentForApplication(userId, appId,
+                                IdentityTenantUtil.getTenantId(user.getTenantDomain()),
+                                userApprovedScopes, null);
+                    } else {
+                        oAuth2ScopeService.addUserConsentForApplication(userId, appId,
+                                IdentityTenantUtil.getTenantId(user.getTenantDomain()),
+                                userApprovedScopes, null);
+                    }
+                }
+            }
+        } catch (IdentityOAuthAdminException e) {
+            throw new OAuthSystemException(
+                    "Error occurred while removing OIDC scopes from disapproved OAuth scopes.", e);
+        } catch (IdentityOAuth2ScopeException e) {
+            throw new OAuthSystemException("Error occurred while storing OAuth scope consent.", e);
+        }
+    }
+
+    private static List<String> getAllowedOAuthScopes(OAuth2Parameters params) throws OAuthSystemException {
+
+        Set<String> allowedScopes = params.getScopes();
+        List<String> allowedOAuthScopes = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(allowedScopes)) {
+            try {
+                startTenantFlow(params.getTenantDomain());
 
             /* If DropUnregisteredScopes scopes config is enabled
              then any unregistered scopes(excluding internal scopes
              and allowed scopes) is be dropped. Therefore they will
              not be shown in the user consent screen.*/
-            if (oauthServerConfiguration.isDropUnregisteredScopes()) {
-                if (log.isDebugEnabled()) {
-                    log.debug("DropUnregisteredScopes config is enabled. Attempting to drop unregistered scopes.");
+                if (oauthServerConfiguration.isDropUnregisteredScopes()) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("DropUnregisteredScopes config is enabled. Attempting to drop unregistered scopes.");
+                    }
+                    allowedScopes = dropUnregisteredScopes(params);
                 }
-                allowedScopes = dropUnregisteredScopes(params);
+
+                // Get registered OIDC scopes.
+                String[] oidcScopes = oAuthAdminService.getScopeNames();
+                List<String> oidcScopeList = new ArrayList<>(Arrays.asList(oidcScopes));
+                for (String scope : allowedScopes) {
+                    if (!oidcScopeList.contains(scope)) {
+                        allowedOAuthScopes.add(scope);
+                    }
+                }
+            } catch (IdentityOAuthAdminException e) {
+                throw new OAuthSystemException("Error while retrieving OIDC scopes.", e);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
             }
+        }
+        return allowedOAuthScopes;
```

Comment:
Lets add a debug log to see the final set of scopes returned.

ex: log.debug("Allowed scopes: " + allowedOAuthScopes + " for request from client: " + params.getClientId());

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1548/commits/27ddbc8fed9c4368ffe7a826b857537a6b8ffff0